lib/lightning_web/live/credential_live/form_component.ex

```diff
@@ -169,20 +140,6 @@ defmodule LightningWeb.CredentialLive.FormComponent do
           schema_changeset: schema_changeset,
           changeset: changeset
         )
-
```

Comment:
The purpose of the tuple `{schema, schema_type}` was to avoid loading the schema from disk with every event.

I see you're no longer using the `schema`, so we need to either remove that from the case statement - or we change this around to only reload the schema when the schema type is changed.

If this feels like too much effort (like will require a whole boat load of logic) then we can leave it and wrap this in a schemaless changeset in the future as an optimization.

Reply:
Mhm @stuartc thanks for explaning the purpose of `{schema, schema_type}`. I did remove it because I couldn't see it's value but now I do understand and kind of wanna put it back. But what would you really suggest me to do there ? I know that we have multiple ways we can do this but what do you think would be the best ?

Reply:
So the idea is that we create a new 'schema' (`Lightning.Credentials.Schema`) when we have a schema type value that we need a schema for - but only if we don't already have a schema _or_ the schema we have is different to the last one.

Currently the `Schema` struct has no information about it's name etc. So we don't have a simple way to just compare what is loaded and what is requested.
I think the way I did it before only worked because only one of the schema types actually had a a schema.

I guess my option to just change the `{_schema, schema_type}` to `schema_type`, and load it every time was because I haven't taken enough time to think about a more robust solution - I feel like avoiding really simple "if before != current" kinda logic.

So I guess for now, lets remove the leftovers as to not confuse someone else and we come back to it. Again, only because I don't have a solid solution right now; and we _will_ be coming back to this when we add more schemas...

Reply:
@stuartc this is now done. Codecov is still complaining tho but I don't see the reason. So tell me if this is ready to merge or if I still need to address something. Thanks